drivers/adc-dac/ad5592r/ad5592r.c

```diff
@@ -91,23 +91,29 @@ int32_t ad5592r_write_dac(struct ad5592r_dev *dev, uint8_t chan,
 }
 
 /**
- * Read ADC channel.
+ * Read ADC.
  *
  * @param dev - The device structure.
- * @param chan - The channel number.
- * @param value - ADC value
+ * @param chans - The ADC channels to be readback
+ * @param values - ADC value array
  * @return 0 in case of success, negative error code otherwise
  */
-int32_t ad5592r_read_adc(struct ad5592r_dev *dev, uint8_t chan,
-			 uint16_t *value)
+int32_t ad5592r_read_adc(struct ad5592r_dev *dev, uint16_t chans,
```

Comment:
i would leave this function to only read one channel. To approach that was used before to add another function for multi channel read is the way to go. The only thing that was wrong about that was that it had multiple spi_reads instead of only one.